build.gradle.kts

```diff
@@ -38,6 +38,11 @@ buildscript {
 }
 
 apiValidation {
+    ignoredPackages.addAll(
+        setOf(
+            "io.sentry.android.core.internal"
```

Comment:
@marandaneto I moved the internal classes under the `internal` package and excluded them from the Api validation, how does that sound?

Reply:
mmm, What's the reason to do that? it's fine for me but trying to understand it first.
Now we're using the `@ApiStatus.Internal` annotation + internal package + excluded from apiValidation, sounds like we have to chose a single approach at least.

Reply:
So, the internal package is needed to exclude from apiValidation conveniently (we just specify a package, instead of a list of classes). 

I thought if the classes are internal, they should not be part of the `.api` files - the `@ApiStatus.Internal` has no effect on this, and is just there to indicate for ourselves/users that the class shouldn't be used outside sentry SDK (though the binary compat validator still checks for those classes)

Reply:
> So, the internal package is needed to exclude from apiValidation conveniently (we just specify a package, instead of a list of classes).

ok, that makes sense.
maybe @bruno-garcia would like to double-check, but I buy it.

Reply:
@bruno-garcia 10% discount for you to buy it

Reply:
wfm: 💵